medcat/cdb.py

```diff
@@ -371,23 +372,28 @@ def update_context_vector(self,
             # Increase counter only for positive examples
             self.cui2count_train[cui] += 1
 
-    def save(self, path: str) -> None:
+    def save(self, path: str, json_path: Optional[str] = None, overwrite: bool = True) -> None:
```

Comment:
Given we don't compress the output json, I'm not surprised its larger. Some interesting results of compression / time taken [here](https://www.lucidchart.com/techblog/2019/12/06/json-compression-alternative-binary-formats-and-compression-methods/).
In another ticket could you add:
- a named arg for compression (default False for the time being) - once JSON compression is the default we can make compression to True also.
- a route to moving away from Dill in major release. i.e. a deprecated warning if dill is used in the next patch release

Reply:
We do compress the overall output before sharing the model. And after that compression, the file sizes are the same (see `Size when packed` above).

While I've not tried any of the methods you've linked, specifically, I think they would sort of defeat the purpose. It's always going to be a balance between size on disk and load time. We can't really expect to lower both (significantly) at the same time when saving the same data.
So if we added compression to the json parts, the load/save times would increase. It's possible we'd still have a net positive, but I'm not convinced it's worth it.

I did at first think that there's a lot that cannot be easily json-serialised. And while there are a few things, they're not actually too many to manage:
 - config.linking.weighted_average_function
   - It's a `partial` so needs some encoding and decoding
 - config.word_skipper
   - It's a `Pattern` so needs some encoding and decoding
 - config.punct_checker
   - `Pattern`
 - vocab.unigram_table
   - It's an `np.ndarray` so needs some encoding and decoding 
 - vocba.vocab
   - Contains `np.ndarray` as each words vectors

But all of that should absolutely be manageable. I'll add a subtask.